morepath/tests/test_security.py

```diff
@@ -81,13 +79,119 @@ def remember(self, response, request, identity):
         def forget(self, response, request):
             pass
 
+    app.commit()
+
     c = Client(app())
 
     response = c.get('/foo')
     assert response.body == b'Model: foo'
     response = c.get('/bar', status=403)
 
 
+def test_permission_directive_identity_app_arg():
+    class App(morepath.App):
+        pass
+
+    class Model(object):
+        def __init__(self, id):
+            self.id = id
+
+    class Permission(object):
+        pass
+
+    @App.verify_identity()
+    def verify_identity(identity):
+        return True
+
+    @App.path(model=Model, path='{id}',
+              variables=lambda model: {'id': model.id})
+    def get_model(id):
+        return Model(id)
+
+    @App.permission_rule(model=Model, permission=Permission)
+    def get_permission(identity, model, permission):
+        if model.id == 'foo':
+            return True
+        else:
+            return False
+
+    @App.view(model=Model, permission=Permission)
+    def default(self, request):
+        return "Model: %s" % self.id
+
+    @App.identity_policy()
+    class IdentityPolicy(object):
+        def identify(self, app, request):
+            assert isinstance(app, App)
+            return Identity('testidentity')
+
+        def remember(self, app, response, request, identity):
```

Comment:
The ``app`` argument here (and in the ``forget`` and ``identify`` methods) is redundant.  I understand that you decided to introduce it here for consistency with other dispatched methods.  The problem with that reasoning is that ``morepath.App.remember_identity`` being a dispatch_method is an implementation detail (that it is so is proven by my ``decoupling`` branch that implements the same API). 

The only thing that the backend developer using Morepath sees is the ``@App.identity_policy`` directive, which is very different from other directives like ``@App.load_json``: it does not decorate a function that is invoked per request on demand, but a callable that returns an ``IdentityPolicy`` object, and that callable is called always exactly once at commit.

So we are dealing here with a completely different beast: painting the claws of a tiger and the hoofs of a sheep the same colour does not make handling them any closer.